packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart

```diff
@@ -812,15 +766,15 @@ class WebViewClient extends JavaObject {
   /// embedded frame changes, i.e. clicking a link whose target is an iframe, it
   /// will also not be called for fragment navigations (navigations to
   /// #fragment_id).
-  void onPageStarted(WebView webView, String url) {}
+  final void Function(WebView webView, String url)? onPageStarted;
```

Comment:
Wouldn't having these fields marked as `final` result into issues when implementing the `PlatformNavigationDelegate` interface from the `webview_flutter_platform_interface` package? 

The `PlatformNavigationDelegate` allows developers to individually register callback handlers through setters (e.g. `PlatformNavigationDelegate.setOnPageStarted(void Function(String url) callback)`). This would suggest developers can update the callback registered on the `PlatformNavigationDelegate` instance any time they want and not only upon construction of the `PlatformNavigationDelegate`. 

Since the `PlatformNavigationDelegate` implementation depends on the `WebViewClient` it somehow needs to be able to construct the instance and pass along the required callback methods when they are set. Meaning this would probably mean we'd would have to construct a new `WebViewClient` instance each time a developer calls one of the setters of the `PlatformNavigationDelegate` implementation.

Reply:
Making these `final` is unfortunately a requirement to deallocate resources automatically. I wanted that solution too, but this is the simplest working solution that I've found so far. For more details you can take a look at https://docs.google.com/document/d/1wXB1zNzYhd2SxCu1_BK3qmNWRhonTB6qdv4erdtBQqo/edit?usp=sharing&resourcekey=0-WOBqqOKiO9SARnziBg28pg

For iOS, I had to cache the callback values in the implementation: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/lib/src/v4/src/webkit_webview_controller.dart#L583. Then the callback methods could just use these values.

Reply:
That would be a good solution, will implement it in a similar way.